### DISCUSSION TITLE
TIFR3 seems to be = 15 permanently, can not reset

### DISCUSSION THREAD

**Discussion by inductosyn** [dbuezas/lgt8fx#318] [Q&A]
.. even immediately after i set TIFR3 = 0 ;
So, how can i use ISR instruction to make a variable frequency generator?
If i set TCCR3A at random moment there are a possibility to have TCNT3 > TCCR3A  even i try to decrement TCCR3A  with 1, so, there will be a period with a wrong length.
I trying to use ISR to modify TCCR3A  at TCNT3 = 0 or closer (i have a minimum value of TCCR3A  anyway, so let it be 10 or even 100), but...

**Comment by dbuezas** [comment]
To clear interrupt flags in TIFR3 you have to set them to 1, not zero.
Try setting up your timer and interrupts with this tool: https://dbuezas.github.io/arduino-web-timers/#mcu=LGT8F328P&timer=3&topValue=ICR3&ICR3=41703&InterruptOnTimerOverflow=on&CompareOutputModeB=clear-up%2C+set-down&OCR3B=20986

It will help you getting the registers right.
